python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -91,10 +91,8 @@ def directed_hausdorff_distance(xs, ys, space_offsets):
         ys,
         as_column(space_offsets, dtype="uint32"),
     )
-    with cudf.core.buffer.acquire_spill_lock():
-        result = result.data_array_view(mode="read")
-        result = result.reshape(num_spaces, num_spaces)
-        return DataFrame(result)
+
+    return DataFrame._from_columns(result, range(num_spaces)).T
```

Comment:
Do you know if transpose is still considered to be unadvised? It used to be a substantial performance hit, maybe the transpose could be performed via thrust instead of here in python?

Reply:
It's probably still a performance hit. The kernel is (unfortunately written) in transposed order. I'll have to modify the kernel to avoid that.

Reply:
Would you say I should change the c++ kernel to avoid breaking python interface, or to keep as is and break the python interface?

Reply:
I'm inclined to break the python interface, the transposition of the result doesn't matter at all iirc.

Reply:
It's a square matrix with i,j or j,i, right?

Reply:
Yes, it will be a change in the order of the output coordinate.

Reply:
At this stage it's still OK to break either API, as long as we tag it appropriately.

Reply:
Transpose is removed.